### PR TITLE
Add knit_print method for webshot

### PR DESCRIPTION
**Pull request by jimhester**
This lets you include `webshot()` calls directly in knitted documents and automatically show the output image.

The object must be returned visibly in order to show up when knitting, so I had to define a print method as well to preserve the previous behavior.

### CODE REVIEW

**Comment by wch** [comment]
I think you can avoid importing knitr. See https://github.com/ramnathv/htmlwidgets/blob/master/R/knitr-methods.R for an example.

**Comment by jimhester** [comment]
I actually had seen that code when I was doing this originally, thought you might not want the added complexity.

https://github.com/wch/webshot/pull/27/commits/8ae7e27883e16c78c8a5f8dab1f53bc3a220b168 keeps knitr in Suggests and uses the referenced functions.

File: NAMESPACE [hunk 1]
```diff
@@ -2,13 +2,12 @@
 
 S3method(appshot,character)
 S3method(appshot,shiny.appobj)
-S3method(knit_print,webshot)
 S3method(print,webshot)
 export("%>%")
 export(appshot)
 export(install_phantomjs)
+export(knit_print.webshot)
```
Comment (yihui) [inline NAMESPACE]: I guess this does not need to be exported.

**Comment by wch** [comment]
Could you also update the functions in image.R to check for the webshot class and also return a filename with that class? Otherwise, I think that doing something like `webshot() %>% resize()` in a knitr document won't result in the image being put inline.

**Comment by jimhester** [comment]
@wch Ok both shrink and resize now also return webshot objects as well.

**Comment by wch** [comment]
Thanks!

**Pull request merged**